pre_commit_dbt/check_macro_is_referenced.py

```diff
@@ -0,0 +1,66 @@
+import argparse
+from typing import Any
+from typing import Dict
+from typing import Optional
+from typing import Sequence
+
+from pre_commit_dbt.utils import add_filenames_args
+from pre_commit_dbt.utils import add_manifest_args
+from pre_commit_dbt.utils import get_json
+from pre_commit_dbt.utils import get_macro_sqls
+from pre_commit_dbt.utils import get_macros
+from pre_commit_dbt.utils import JsonOpenError
+
+def get_macro_name(macros_ref):
+    return map(lambda x:x.split(".")[-1], macros_ref) 
+
+def get_macros_in_models(manifest):
+    """
+    Gets macros listed in the "depends_on" key of the "nodes" key in the manifest
+    Arguments:
+        manifest: manifest file 
+    Returns:
+        macros: list of macros
+    """
+    macros = []
+    nodes = manifest.get("nodes", {})
+    for key,value in nodes.items() :
+            macros_ref_in_models = value.get("depends_on",{}).get("macros",{})
+            macro_ref_name = get_macro_name(macros_ref_in_models) 
+            macros.extend(macro_ref_name)
+    return macros
+
+def is_referenced(paths: Sequence[str], manifest: Dict[str, Any]) -> int:
```

Comment:
IMO the name does not cover well the goal of the function. `is_referenced` would be for a single item. But here, we iterate over all macros. So we need to say that explicitely in the name. 

Same goes for the hook name. This does not represent the code behaviour.

Reply:
I agree with Benoît Sensei.